Add uploading OVN backups to Ceph Swift API gateway

awfabian-rs opened the pull request:
JIRA:OSPC-341

awfabian-rs commented:
Current in draft status to:

1. See how the _CronJob_ runs on my test cluster when the cron job time happens.
    - which I will check tomorrow
2. Look at https://github.com/rackerlabs/genestack/pull/151 as an example and see if we can use _Vault_ here.

cloudnull commented on docs/infrastructure-ovn-setup.md:
```diff
@@ -98,3 +98,11 @@ kubectl apply -k /opt/genestack/kustomize/ovn
+
+!!! note
+
+    To upload backups to OSPCv1 Cloud Files, edit ovn-backup-config-patch.yaml
```
This is great. Would this work for v2 cloudfiles that is being built in SJC?

awfabian-rs commented:
- I modified this to use a Ceph Swift API gateway because we do our SJC database MariaDB backups to that.
- I tested this on the DFW environment.
- I moved a few things out into separate files and used some _kustomize_ generators
    - It seems easier to read `ovn-backup.sh` as a stand-alone file.
    - The `ovn-backup-config` patch almost completely duplicated the `ConfigMap` in the main file, and it just sets environment variables anyway, so I used a _kustomize_ `ConfigMap` generator to just put the key-value pairs into a `ConfigMap`.
    
Feel free to merge if it looks ready to you. Please review and let me know any change(s) needed.
